#Measures of Central Tendency and Dispersion

In [13]:

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


In [17]:
df=pd.read_csv('studentsPerformance.csv')
df

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

1. Find out how many males and females participated in the test.


In [6]:
print(list(df['gender']).count('female'))

518


In [7]:
print(list(df['gender']).count('male'))

482


No of females participated in the test = 518
No of males participated in the test= 482

2.What do you think about the students' parental level of education

In [19]:
df.mode()

gender race/ethnicity parental level of education     lunch  \
0  female        group C                some college  standard   

  test preparation course  math score  reading score  writing score  
0                    none          65             72             74

Most of the parents are educated from some college

3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course

In [18]:
df[['test preparation course',
      'gender',
      'math score',
      'writing score',
      'reading score']].groupby(['test preparation course','gender']).agg('median')

math score  writing score  reading score
test preparation course gender                                          
completed               female          67             79             78
                        male            74             71             72
none                    female          62             70             71
                        male            67             60             63

4. What do you think about the scoring variation for math, reading and writing
based on
● Gender
● Test preparation course

Females who have completed the test preparation course has scored more marks on reading and writing,while males with completed course, scored more on maths.
In both the cases, males have scored high on mathematics and females on reading and writing.

5. The management needs your help to give bonus points to the top 25% of
students based on their maths score, so how will you help the management
to achieve this.

In [25]:
bonus=np.quantile(df["math score"],0.25)

df["bonus score"]=[0 if i<bonus else 5 for i in df["math score"]]

In [27]:
df[df["bonus score"]==5]

gender race/ethnicity parental level of education         lunch  \
153  female        group B          associate's degree      standard   
154  female        group B            some high school      standard   
155  female        group C          associate's degree  free/reduced   
156  female        group C          associate's degree      standard   
157  female        group D          associate's degree  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  \
153                    none          57             69             68   
154                    none          57             67             72   
155                    none          57             78             67   
156               completed          57             77             80   
157               completed          57             74             76   
..                      ...         ...            ...            ...   
995               completed          99             87             81   
996               completed         100             96             86   
997               completed         100             97             99   
998               completed         100            100             93   
999               completed         100            100            100   

     bonus score  
153            5  
154            5  
155            5  
156            5  
157            5  
..           ...  
995            5  
996            5  
997            5  
998            5  
999            5  

[753 rows x 9 columns]